File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/front/top/TopViewModel.java
package com.ikmr.banbara23.yaeyama_liner_checker.front.top;

import android.databinding.ObservableBoolean;
import android.databinding.ObservableField;
import android.databinding.ObservableInt;

import com.ikmr.banbara23.yaeyama_liner_checker.model.top.TopPort;

/**
 * Top画面のViewModel
 */
public class TopViewModel {
    // ステータス
//    public ObservableField<TopCompanyInfo> topCompany = new ObservableField<>();
    // 港別ステータス
    public ObservableField<TopPort> topPort = new ObservableField<>();
    public ObservableBoolean showPortProgress = new ObservableBoolean();

    // 天気
    public ObservableField<String> todayWeather = new ObservableField<>();
    public ObservableBoolean showWeatherProgres = new ObservableBoolean();

    // 会社別ステータス
    public ObservableInt aneiColor = new ObservableInt();
    public ObservableInt ykfColor = new ObservableInt();
    public ObservableInt dreamColor = new ObservableInt();

    public ObservableField<String> aneiStatus = new ObservableField<>();
    public ObservableField<String> ykfStatus = new ObservableField<>();
    public ObservableField<String> dreamStatus = new ObservableField<>();

    public ObservableBoolean showCompanyProgres = new ObservableBoolean();
}

File: app/src/main/java/com/ikmr/banbara23/yaeyama_liner_checker/front/top/TopViewModel.kt
package com.ikmr.banbara23.yaeyama_liner_checker.front.top

import android.graphics.drawable.Drawable
import androidx.databinding.ObservableBoolean
import androidx.databinding.ObservableField
import com.ikmr.banbara23.yaeyama_liner_checker.model.top.TopPort

/**
 * Top画面のViewModel
 */
class TopViewModel {

    // 港別ステータス
    var topPort = ObservableField<TopPort>()
    var showPortProgress = ObservableBoolean()

    // 天気
    var date = ObservableField<String>()
    var weather = ObservableField<String>()
    var showWeatherProgress = ObservableBoolean()

    // 会社別ステータス
    // var aneiColor = ObservableInt()
    // var ykfColor = ObservableInt()

    var aneiStatus = ObservableField<String>()
    var ykfStatus = ObservableField<String>()

    var showCompanyProgress = ObservableBoolean()

    // 台風
    var typhoon = ObservableField<String>()
    var showTyphoonProgress = ObservableBoolean()

    var aneiBackground = ObservableField<Drawable>()
    var ykfBackground = ObservableField<Drawable>()
}
